transcode/transcode.go

```diff
@@ -568,47 +605,57 @@ func transcodeSegment(
 	duration := time.Since(start)
 	metrics.Metrics.TranscodeSegmentDurationSec.Observe(duration.Seconds())
 
-	for _, transcodedSegment := range tr.Renditions {
-		renditionIndex := getProfileIndex(transcodeProfiles, transcodedSegment.Name)
-		if renditionIndex == -1 {
-			return fmt.Errorf("failed to find profile with name %q while parsing rendition segment", transcodedSegment.Name)
+	for renditionIndex, profile := range encodedProfiles {
```

Comment:
same here, pretty simple function but maybe better to break it up into a separate fcn to unit test?